services/blockstorage/test/get_block_headers.go

```diff
@@ -0,0 +1,105 @@
+package test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/orbs-network/orbs-network-go/test"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+)
+
+var _ = Describe("Block storage", func () {
+	When("asked to provide transactions block header", func () {
+		It("returns transactions block header", func () {
+			driver := NewDriver()
+			driver.expectCommitStateDiff()
+
+			block := test.BlockPairBuilder().Build()
+			driver.commitBlock(block)
+
+			output, err := driver.blockStorage.GetTransactionsBlockHeader(&services.GetTransactionsBlockHeaderInput{BlockHeight:1})
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(output.TransactionsBlockHeader).To(Equal(block.TransactionsBlock.Header))
+			Expect(output.TransactionsBlockMetadata).To(Equal(block.TransactionsBlock.Metadata))
+			Expect(output.TransactionsBlockProof).To(Equal(block.TransactionsBlock.BlockProof))
+		})
+
+		It("blocks if requested block is in near future", func (done Done) {
+			driver := NewDriver()
+			driver.expectCommitStateDiff()
+
+			block := test.BlockPairBuilder().Build()
+			driver.commitBlock(block)
+
+			result := make(chan *services.GetTransactionsBlockHeaderOutput)
+
+			go func () {
+				output, _ := driver.blockStorage.GetTransactionsBlockHeader(&services.GetTransactionsBlockHeaderInput{BlockHeight:5})
+				result <- output
+			}()
+
+			for i:=2; i <=6 ; i++ {
+				driver.commitBlock(test.BlockPairBuilder().WithHeight(i).Build())
+			}
+
+			Expect(driver.getLastBlockHeight().LastCommittedBlockHeight).To(Equal(primitives.BlockHeight(6)))
+
+			block5 := driver.getBlock(5).TransactionsBlock
```

Comment:
why do you need the comparison again? the added functionality here is to wait until txBlock is ready, so I'd only check that the call returns something, no need to assert on content again

Reply:
What if it waits and returns the last available block and not the one we requested? Or some other block?

Reply:
cool, but you only need to assert that the block that returned has the expected height, you don't need to assert on content